src/elements/element.line.js

```diff
@@ -48,6 +48,7 @@ class Line extends Element {
 		}
 
 		if (me._loop) {
+			points = points.slice(); // clone array
```

Comment:
great catch! I had noticed this clone taking awhile, but didn't see that we could move it to occur here